### PR TITLE
[Feature:TAGrading] NavigationBar for grading_beta

### PR DESCRIPTION
**Pull request by NiaHeermance**
### Dependencies
This PR is a branch of PR #9454, and thus PR #9454 should be merged first.

### What is the new behavior?
The grading_beta page now has the NavigationBar. The Home button takes one back to the Details page. The Previous Student and Next Student buttons do nothing currently. (That is for the next PR.) The full-screen button works as expected. The settings button does nothing currently, and the Progress Bar currently is fixed at 65%.

### What to test?
- The Home button on grading_beta does indeed take you to the Gradeable's details page.

- The Full Screen button on grading_beta does indeed fullscreen the page.

- If Anonymous Mode is enabled as an instructor on the Details page, the grading_beta does not display the Student Information on the left side of the navigation bar. 
    - If Anonymous Mode is disabled, the Student Informtaion on the left side of the navigation bar is visible.

- If you copy the current URL of a grading_beta page to a browser that is logged in as student, the Student Information on the left side of the navigation bar should not appear.

### How do I access grading_beta?
1. On the main course's page, pick the Grade button for an assignment on the Gradeable page

image

2. You are now on the Gradeable's Details page. Click the Grade/Score button for a particular student, and if this Gradeable is an Electronic/Rubric (Rubric is the new name we are changing to) gradeable, the TA Grading page will appear.

image

3. The URL will look something like this:

SERVER/courses/TERM/COURSE/gradeable/GRADEABLE_NAME/grading/grade?who_id=ANONYMOUS_STUDENT_HASH&sort=id&direction=ASC

Change /grading/ in the URL to /grading_beta/ to open grading_beta.

image

4. grading_beta will look like the following:

image

### CODE REVIEW

**Comment by williamjallen** [comment]
Can you please add some screenshots to your PR description to make it easier for reviewers to see what the intended behavior is here?

File: site/app/views/grading/popup_refactor/RubricGraderView.php [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+     * @param Gradeable $gradeable - The current gradeable.
+     * @param GradedGradeable $current_submission - The current submission we are looking at.
+     * @param Gradeable $gradeable - The current Gradeable.
+     * @param string $sort_type - The current way we are sorting students. Used to create the header.
+     * @param string $sort_direction -  Either "ASC" or "DESC" for ascending or descending sorting order.
+     *     Used to create the header.
+     * @param bool $is_peer_gradeable - True if the gradeable has peer grading.
+     * @param bool $is_team_gradeable - True if the gradeable is a team gradeable.
+     * @param string $blind_access_mode - Either "unblind", "single", or "double". See above for details.
+     * @param string $details_url - URL of the details page for this Gradeable.
```
Comment (williamjallen) [inline site/app/views/grading/popup_refactor/RubricGraderView.php:89]: It would be good to add native PHP type annotations to all of the functions in this PR.  Given that you have provided a description of each parameter, it would be fine to have both PHPDoc annotations and native PHP type annotations.

File: site/app/views/grading/popup_refactor/RubricGraderView.php [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+                    <div class="content-item content-item-right">
+                HTML;
+
+        $page_html .= $this->renderNavigationBar($details_url);
+
+        return $page_html . <<<HTML
+                                         </div>
+                                     </div>
+                                </div>
+            HTML;
```
Comment (williamjallen) [inline site/app/views/grading/popup_refactor/RubricGraderView.php:121]: The indentation is pretty weird here.  In reality, the indentation of the final HTML sent to the browser doesn't matter, so it would be fine to just return the string `</div></div></div>`.  I do question why we are manipulating HTML in PHP in the first place though.  This kind of stuff is exactly why we have Twig.

File: site/public/js/ta-grading.js [hunk 3]
```diff
@@ -592,8 +592,8 @@ function updateCookies(){
 // Student navigation
 function gotoMainPage() {
 
-  let window_location = $("#main-page")[0].dataset.href
-
+  let window_location = $("#main-page")[0].dataset.href;
+
```
Comment (williamjallen) [inline site/public/js/ta-grading.js:596]: This change seems unrelated to the rest of this PR so it would probably be best to revert it to prevent accidental merge conflicts.

**Review by williamjallen — CHANGES_REQUESTED** [review]
Here are my comments from an initial pass through the code.  I'll take a deeper look later and provide some more feedback.

File: site/app/views/grading/popup_refactor/RubricGraderView.php (hunk 1 above)
Comment (NiaHeermance) [reply site/app/views/grading/popup_refactor/RubricGraderView.php:89]: I'm not really sure what you mean. Aren't those annotations above?

File: site/app/views/grading/popup_refactor/RubricGraderView.php (hunk 2 above)
Comment (NiaHeermance) [reply site/app/views/grading/popup_refactor/RubricGraderView.php:121]: I got this string from the original ElectronicGraderView.php. I just fixed the indenting. I could totally move this string to its own Twig file, but if I did that, I would have to call a Twig file within a twig file. Do you have a good link of an example of the syntax of this?

**Comment by NiaHeermance** [comment]
> Can you please add some screenshots to your PR description to make it easier for reviewers to see what the intended behavior is here?

Done!

File: site/app/views/grading/popup_refactor/RubricGraderView.php (hunk 2 above)
Comment (williamjallen) [reply site/app/views/grading/popup_refactor/RubricGraderView.php]: Discussed in person before the meeting.

File: site/app/views/grading/popup_refactor/RubricGraderView.php (hunk 1 above)
Comment (williamjallen) [reply site/app/views/grading/popup_refactor/RubricGraderView.php:89]: Oops, my bad.  I completely missed that there were already PHP type annotations.  🤦‍♂️

**Review by 04vmatsibekker — APPROVED** [review]

File: site/app/controllers/grading/popup_refactor/RubricGraderController.php [hunk 4]
```diff
@@ -152,4 +203,111 @@ private function setCurrentGradeable(string $gradeable_id) {
...
+        return $this->core->buildCourseUrl(['gradeable', $this->gradeable->getId(), 'grading', 'details'])
+            . '?'
+            . http_build_query(['sort' => $this->sort_type, 'direction' => $this->sort_direction]);
+    }
+
+
+    /**
+     * Sets the current user type for the website user.
+     */
+    private function setUserType() {
```
Comment (shailpatels) [inline site/app/controllers/grading/popup_refactor/RubricGraderController.php:241]: similar comment about `$user_type` vs `$group` naming as above

File: site/app/controllers/grading/popup_refactor/RubricGraderController.php [hunk 5]
```diff
@@ -87,18 +131,22 @@ public function createMainRubricGraderPage(
         string $direction = "ASC",
         string $navigate_assigned_students_only = "true"
     ) {
-
-        $this->setMemberVariables($gradeable_id, $who_id, $sort, $direction, $navigate_assigned_students_only);
+        $this->setMemberVariables($gradeable_id, $who_id, $sort, $direction);
```
Comment (shailpatels) [inline site/app/controllers/grading/popup_refactor/RubricGraderController.php:134]: `$this->setMemberVariables` calls the `setCurrentSubmission` function which can possibly set `$this->current_submission` to false if it fails to get the gradeable for whatever reason. Even though it attempts to add a redirect to the page, the renderOutput function is still calling the `createRubricGradeableView` which is expecting a GradedGradeable for the $current_submission variable but could possibly be false

**Review by shailpatels — CHANGES_REQUESTED** [review]

File: site/app/views/grading/popup_refactor/RubricGraderView.php [hunk 6]
```diff
@@ -15,60 +14,133 @@
...
+
+
+    // ---
+
+
+    // ---------------------------------
 
     /**
      * Creates the Rubric Grading page visually.
      * This function is called in reateMainRubricGraderPage of RubricGraderController.php.
```
[the remark this answers is not in the document]
Comment (NiaHeermance) [reply site/app/views/grading/popup_refactor/RubricGraderView.php]: Thanks!

File: site/app/views/grading/popup_refactor/RubricGraderView.php [hunk 7]
```diff
@@ -15,60 +14,133 @@
...
      * The current gradeable being graded.
      */
     private $gradeable;
 
-    # ---------------------------------
+    /**
+     * @var GradedGradeable
+     * The current submission being graded.
+     */
+    private $current_submission;
```
[the remark this answers is not in the document]
Comment (NiaHeermance) [reply site/app/views/grading/popup_refactor/RubricGraderView.php:42]: That's a good point. I'm going to leave it as is for now but I'll keep that in mind for the future.

[the remark this answers is not in the document]
Comment (NiaHeermance) [reply site/app/views/grading/popup_refactor/RubricGraderView.php]: Update: I understand now what you mean. I didn't realize PHP doesn't keep state between server requests. I've now deleted all the member variables as they were unnecessary.
